src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+import { ErrorMessage } from '../types/ErrorMessage';
+
+type Props = {
+  inputRef: React.MutableRefObject<HTMLInputElement | null>;
+  onAddTodo: (title: string) => Promise<void>;
+  error: ErrorMessage;
+  setErrorMessage: (error: ErrorMessage) => void;
+  isInputDisablet: boolean;
+  isDeletedTodos: number[];
+};
+
+export const TodoHeader: React.FC<Props> = props => {
+  const {
+    inputRef,
+    onAddTodo,
+    error,
+    setErrorMessage,
+    isInputDisablet,
+    isDeletedTodos,
+  } = props;
+
+  const [title, setTitle] = useState('');
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputRef, isInputDisablet, isDeletedTodos]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (error) {
+      setErrorMessage(ErrorMessage.Default);
+    }
+
+    if (!title.trim()) {
+      setErrorMessage(ErrorMessage.EmptyTitle);
+
+      return;
+    }
+
+    try {
+      await onAddTodo(title.trim());
+      setTitle('');
+    } finally {
+    }
```

Comment:
```suggestion
    await onAddTodo(title.trim());
    setTitle('');
```
You don't need finally if you do nothing.